### PR TITLE
feat(add method is_private_use1() in class Device)

### PR DESCRIPTION
**Pull request by ykddd**
As the title

### CODE REVIEW

**Comment by ykddd** [comment]
@bdhirsh Hi，I'm a newbie, and I'd like to make a little contribution in completing the PrivateUse1 mechanism.I am looking forward to your review, thank you~

**Review by bdhirsh — APPROVED** [review]
sure, this seems ok. Thanks!

File: c10/core/Device.h [hunk 1]
```diff
@@ -146,6 +146,11 @@ struct C10_API Device final {
     return type_ == DeviceType::CPU;
   }
 
+  /// Return true if the device is of PrivateUse1 type.
+  bool is_private_use1() const noexcept {
```
Comment (bdhirsh) [inline c10/core/Device.h:150]: nit: can you just make this `is_privateuse1` to reflect the pattern for other variants of this method?

Comment (ykddd) [reply c10/core/Device.h]: Good advice,I have already modified it,please review it again.

> nit: can you just make this `is_privateuse1` to reflect the pattern for other variants of this method?

**Pull request closed without merging**